### ISSUE TITLE
A multi-branch 3D ResNet based network, an ambigous global loss 

### ISSUE THREAD

**Issue by Emnaa12** [JihongJu/keras-resnet3d#7]
Hi @JihongJu 
Thank you for this repository, 
I have tried to conceive a multi-branch network consisting of two branches, the first is a 3D ResNet50 and the second is 2D ResNet50. However, the global loss which is theorically a weighted summation of two partial losses, is much higher than the weighted summation. I have tested with other networks and it works correctly. Unfortunately, it is not the case with this code. Here is my code 
https://github.com/Emnaa12/Multi-branch-CNN
and here some training epoch results:

    Epoch 2/200
    26/26 [==============================] - 39s 1s/step - loss: 9.2902 - 
    dense_1_loss: 0.0801 - dense_2_loss: 0.0717 -
    Epoch 3/200
    26/26 [==============================] - 39s 1s/step - loss: 8.2261 - 
    dense_1_loss: 0.0251 - dense_2_loss: 0.0199 -
    Epoch 4/200 
    26/26 [==============================] - 39s 2s/step - loss: 7.3107 - 
    dense_1_loss: 0.0595 - dense_2_loss: 0.0048 

Could you please have a look to my code and figure out the key problem of this issue?

**Comment by AlmCoding** [comment]
I have had the issue that the loss and metric were not the same in my two-trunk network (3d-resnet for video and 2d-resnet for audio). After I removed l2 regularization from the Conv3d layers my loss function (MAE) matches the MAE metric as its supposed to. This may help others as well...
Any guess why the l2 regularization causes the loss differ from the metric while sharing the same function?

**Comment by Emnaa12** [comment]
@AlmCoding exactly the same! I have resolved the issue through removing L2 regularisation option from 3D ResNet.
